From f181bfa2e98557556117aab9dec7c163747efb87 Mon Sep 17 00:00:00 2001
From: "Ruslan, Rusaimi Amira" <rusaimi.amira.ruslan@intel.com>
Date: Fri, 13 Apr 2018 17:49:19 +0800
Subject: [PATCH 10/43] misc: intel-ish-clients: Remove linux header file for
 uapi

Remove linux mutex and miscdevice header file, should not expose to userspace.
Get compile error when CONFIG_HEADERS_CHECK is enable.

Signed-off-by: Ruslan, Rusaimi Amira <rusaimi.amira.ruslan@intel.com>
Signed-off-by: Lay, Kuan Loon <kuan.loon.lay@intel.com>
---
 drivers/misc/intel-ish-client/intel-ishtp-clients.c | 1 +
 include/uapi/linux/intel-ishtp-clients.h            | 2 --
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/misc/intel-ish-client/intel-ishtp-clients.c b/drivers/misc/intel-ish-client/intel-ishtp-clients.c
index 3d8de4b..cee855e 100644
--- a/drivers/misc/intel-ish-client/intel-ishtp-clients.c
+++ b/drivers/misc/intel-ish-client/intel-ishtp-clients.c
@@ -29,6 +29,7 @@
 #include <linux/types.h>
 #include <linux/uuid.h>
 #include <linux/uaccess.h>
+#include <linux/mutex.h>
 
 #include "ishtp-dev.h"
 #include "client.h"
diff --git a/include/uapi/linux/intel-ishtp-clients.h b/include/uapi/linux/intel-ishtp-clients.h
index d228f19..ac47a55 100644
--- a/include/uapi/linux/intel-ishtp-clients.h
+++ b/include/uapi/linux/intel-ishtp-clients.h
@@ -18,8 +18,6 @@
 #define _INTEL_ISHTP_CLIENTS_H
 
 #include <linux/uuid.h>
-#include <linux/mutex.h>
-#include <linux/miscdevice.h>
 
 /*
  * This IOCTL is used to associate the current file descriptor with a
-- 
1.9.1

